### PR TITLE
Add a dedicated order state for imported orders from shoppingfeed

### PR DESCRIPTION
**Pull request by bogdan202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add a dedicated order state for imported orders from shoppingfeed.
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes 34463
| How to test?  | Please indicate how to best verify that this PR is correct.

### CODE REVIEW

**Review by clotairer — CHANGES_REQUESTED** [review]

File: src/OrderImport/SFOrderState.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use Configuration;
+use Db;
+use DbQuery;
+use Language;
+use OrderState;
+use Validate;
+
+class SFOrderState
+{
+    const SF_ORDER_STATE = 'SHOPPINGFEED_ORDER_STATE';
```
[the remark this answers is not in the document]
Comment (bogdan202) [reply src/OrderImport/SFOrderState.php]: done

File: classes/actions/ShoppingfeedOrderImportActions.php [hunk 2]
```diff
@@ -736,7 +737,7 @@ public function validateOrder()
         try {
             $paymentModule->validateOrder(
                 (int) $cart->id,
-                Configuration::get('PS_OS_PAYMENT'),
+                $this->initSfOrderState()->get()->id,
```
[the remark this answers is not in the document]
Comment (bogdan202) [reply classes/actions/ShoppingfeedOrderImportActions.php:740]: SFOrderState::get() returns SHOPPINGFEED_ORDER_STATE
if it doesn't exist, then SFOrderState creates it
if creating is failed, then it returns PS_OS_PAYMENT

[the remark this answers is not in the document]
Comment (clotairer) [reply classes/actions/ShoppingfeedOrderImportActions.php:740]: the problem is that if a merchant upgrade the module, new order trigger the SHOPPINGFEED_ORDER_STATE state and new order will be imported in this new state. But if he an ERP that use PS_OS_PAYMENT to sync new order, he need to change the configuration somewhere. We don't supply a continuous behaviour and introduce a BC.

**Review by clotairer — CHANGES_REQUESTED** [review]

File: classes/actions/ShoppingfeedOrderImportActions.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (bogdan202) [reply classes/actions/ShoppingfeedOrderImportActions.php:740]: Added the possibility to configure the first order state after an import

**Review by clotairer — CHANGES_REQUESTED** [review]

**Review by clotairer — APPROVED** [review]

**Pull request merged**